### PR TITLE
tests: fix postrm-purge test on systems with sytemd v253

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Move the removal for the $SNAP_MOUNT_DIR dir after the services are removed because those errors are making services fail.

test-snapd-service.test-snapd-endure-service[19138]: error: cannot find installed snap "test-snapd-service" at revision x1: missing file /var/lib/snapd/snap/test-snapd-service/x1/meta/snap.yaml

Also, added a small wait to make sure the units are already stopped before continue.

### CODE REVIEW

**Comment by sergiocazzolato** [comment]
I still see those units active after snapd is removed:

# systemctl -l | grep -E '(snap.|snapd.)'
  lvm2-monitor.service                                                                        loaded    active exited    Monitoring of LVM2 mirrors, snapshots etc. using dmeventd or progress polling
● snap.test-snapd-service.test-snapd-service-refuses-to-stop.service                          not-found failed failed    snap.test-snapd-service.test-snapd-service-refuses-to-stop.service
● snapd.apparmor.service                                                                      not-found active exited    snapd.apparmor.service
● snapd.mounts-pre.target                                                                     not-found active active    snapd.mounts-pre.target
● snapd.mounts.target                                                                         not-found active active    snapd.mounts.target
google:arch-linux-64 .../tests/main/postrm-purge# systemctl status snapd.apparmor.service
● snapd.apparmor.service
     Loaded: not-found (Reason: Unit snapd.apparmor.service not found.)
     Active: active (exited) since Fri 2023-04-21 19:41:23 UTC; 6min ago
   Main PID: 12217 (code=exited, status=0/SUCCESS)
        CPU: 8ms

Apr 21 19:41:23 apr211935-855620 systemd[1]: Starting Load AppArmor profiles managed internally by snapd...
Apr 21 19:41:23 apr211935-855620 snapd-apparmor[12217]: main.go:121: No profiles to load
Apr 21 19:41:23 apr211935-855620 systemd[1]: Finished Load AppArmor profiles managed internally by snapd.
google:arch-linux-64 .../tests/main/postrm-purge# systemctl status snapd.mounts.target
● snapd.mounts.target
     Loaded: not-found (Reason: Unit snapd.mounts.target not found.)
     Active: active since Fri 2023-04-21 19:41:23 UTC; 6min ago

Apr 21 19:41:23 apr211935-855620 systemd[1]: Reached target Mounted snaps.
google:arch-linux-64 .../tests/main/postrm-purge# systemctl status snapd.mounts-pre.target
● snapd.mounts-pre.target
     Loaded: not-found (Reason: Unit snapd.mounts-pre.target not found.)
     Active: active since Fri 2023-04-21 19:41:23 UTC; 7min ago

**Review by mvo5 — APPROVED** [review]
Thank you

File: cmd/snap-mgmt/snap-mgmt.sh.in [hunk 1]
```diff
@@ -27,10 +27,18 @@ SNAP_UNIT_PREFIX="$(systemd-escape -p ${SNAP_MOUNT_DIR})"
...
+    echo "Stopping unit $unit"
+    systemctl stop -q "$unit" || true
+
+    for i in $(seq 20); do
+        echo "Waiting until unit $unit is stopped [attempt $i]"
+        if ! systemctl is-active -q "$unit"; then
+            echo "$unit is stopped."
+            return
+        fi
+        sleep .1
```
Comment (mvo5) [inline cmd/snap-mgmt/snap-mgmt.sh.in:40]: This is quite short (20 * 0.1sec) - should we increase this for slow boards like pi2 etc?

**Review by Meulengracht — APPROVED** [review]
LGTM, but maybe increase the sleep like mvo suggested

File: cmd/snap-mgmt/snap-mgmt.sh.in (hunk 1 above)
Comment (sergiocazzolato) [reply cmd/snap-mgmt/snap-mgmt.sh.in]: updates to (10 * .5). Just a comment, this script shouldn't be used in ubuntu core, it is just for classic.

Comment (sergiocazzolato) [reply cmd/snap-mgmt/snap-mgmt.sh.in]: thanks for the review!!

**Pull request merged**